pkg/operator/ceph/version/version.go

```diff
@@ -98,37 +104,50 @@ func (v *CephVersion) ReleaseName() string {
 // ExtractCephVersion extracts the major, minor and extra digit of a Ceph release
 func ExtractCephVersion(src string) (*CephVersion, error) {
 	var build int
-	m := versionPattern.FindStringSubmatch(src)
-	if m == nil {
+	var buildSuffix string
+	var image string
+	versionMatch := versionPattern.FindStringSubmatch(src)
+	if versionMatch == nil {
 		return nil, errors.Errorf("failed to parse version from: %q", src)
 	}
 
-	major, err := strconv.Atoi(m[1])
+	major, err := strconv.Atoi(versionMatch[1])
 	if err != nil {
-		return nil, errors.Errorf("failed to parse version major part: %q", m[1])
+		return nil, errors.Errorf("failed to parse version major part: %q", versionMatch[1])
 	}
 
-	minor, err := strconv.Atoi(m[2])
+	minor, err := strconv.Atoi(versionMatch[2])
 	if err != nil {
-		return nil, errors.Errorf("failed to parse version minor part: %q", m[2])
+		return nil, errors.Errorf("failed to parse version minor part: %q", versionMatch[2])
 	}
 
-	extra, err := strconv.Atoi(m[3])
+	extra, err := strconv.Atoi(versionMatch[3])
 	if err != nil {
-		return nil, errors.Errorf("failed to parse version extra part: %q", m[3])
+		return nil, errors.Errorf("failed to parse version extra part: %q", versionMatch[3])
 	}
 
 	// See if we are running on a build release
-	mm := buildVersionPattern.FindStringSubmatch(src)
+	buildVersionMatch := buildVersionPattern.FindStringSubmatch(src)
 	// We don't need to handle any error here, so let's jump in only when "mm" has content
-	if mm != nil {
-		build, err = strconv.Atoi(mm[4])
+	if buildVersionMatch != nil {
+		build, err = strconv.Atoi(buildVersionMatch[4])
 		if err != nil {
-			logger.Warningf("failed to convert version build number part %q to an integer, ignoring", mm[4])
+			logger.Warningf("failed to convert version build number part %q to an integer, ignoring", buildVersionMatch[4])
 		}
 	}
 
-	return &CephVersion{major, minor, extra, build}, nil
+	// get build suffix if running on a build release
+	buildSuffixMatch := buildSuffixPattern.FindStringSubmatch(src)
+	if buildSuffixMatch != nil {
+		buildSuffix = buildSuffixMatch[5]
+	}
+
+	imageMatch := imagePattern.FindStringSubmatch(src)
```

Comment:
Why is this needed?

Reply:
trying to fetch the commit ID of the image here. I believe this should address the [commit](https://github.com/rook/rook/pull/8060#pullrequestreview-677678759)